buku

```diff
@@ -553,35 +567,33 @@ class BukuDb:
         Returns
         -------
         int
-            DB index, or -1 if URL not found in DB.
+            DB index, or None if URL not found in DB.
         """
 
-        self.cur.execute('SELECT id FROM bookmarks WHERE URL = ? LIMIT 1', (url,))
-        resultset = self.cur.fetchall()
-        return resultset[0][0] if resultset else -1
+        row = self._fetch_first('SELECT * FROM bookmarks WHERE url = ?', url)
+        return row and row.id
 
     def get_max_id(self) -> int:
         """Fetch the ID of the last record.
 
         Returns
         -------
         int
-            ID of the record if any record exists, else -1.
+            ID of the record if any record exists, else None.
         """
 
-        self.cur.execute('SELECT MAX(id) from bookmarks')
-        resultset = self.cur.fetchall()
-        return -1 if resultset[0][0] is None else resultset[0][0]
+        self.cur.execute('SELECT MAX(id) FROM bookmarks')
+        return self.cur.fetchall()[0][0]
 
     def add_rec(
             self,
             url: str,
             title_in: Optional[str] = None,
             tags_in: Optional[str] = None,
             desc: Optional[str] = None,
-            immutable: Optional[int] = 0,
-            delay_commit: Optional[bool] = False,
-            fetch: Optional[bool] = True) -> int:
+            immutable: int = FLAG_NONE,
```

Comment:
It can and really should in my opinion, but [you shot down this suggestion back when I made it](https://github.com/jarun/buku/issues/651) :sweat_smile:

Reply:
Since you are pointing repeatedly to the fact that you have shared this and I responded some other way - this is a piece of code I wrote years back. I have forgotten the logic and python by now. Nor do I care much about cosmetic changes in a bookmarking utility anymore. I wanted to see all the actual changes in one place to decide what to use and what not to.

Reply:
…As I mentioned before, my original idea was to change it to bool in the API (as it makes more sense to be used this way), but to keep it a bitmask in the DB (to avoid data migration).